ppmclibs/tinycv_impl.cc

```diff
@@ -160,11 +157,11 @@ std::vector<Point> minVec(const Mat& m, float min)
 // Used to sort a list of points to find the closest to the original
 struct SortByClose {
     SortByClose(int _x, int _y) { orig.x = _x, orig.y = _y; }
-    bool operator()(Point a, Point b)
+    bool operator()(const cv::Point& a, const cv::Point& b) const
```

Comment:
`cv::Point` is just two ints so it is as big as a pointer. So I would actually prefer passing them by value.

Reply:
passing by reference generalise better

Reply:
Usually indeed it's opposite - prefer passing by reference over value, the latter only when you don't care about performance / memory, or you actually need a data copy.

Reply:
There's no general 'pass things by reference' rule. The C++ Core Guidelines always distinguish between cheap to copy and ... just have a look at this table: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information

So I was basically just following this guideline: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const

Reply:
That would work, if we were not trying to optimize this code. So 'prefer simple' it's not what i'm looking for in this specific case.

How many points of that guide you think we are following in tinycv?

Reply:
In general on amd64 you are right it is the same size but as you can see below on arm it is not anymore.

   hell@beaglebone:~/test/build$ ./test-p
   Size of cv::Point: 8
   Size of ref to cv::Point: 4

If you are working with the structure before you call this function it is in cache and passing by reference is even more quicker than copy as it is in the cache and processor can work with it fast and lot of other reasons.